### PR TITLE
Improve theming capabilities for Sidebar and Masterbar

### PR DESCRIPTION
**Pull request by frontdevde**
#### Changes proposed in this Pull Request

* Improve theming capabilities for Sidebar and Masterbar
* Update overflow-gradient Mixin
* Update long-content-fade Mixin

#### Testing instructions

* Build the styles, everything should work as expected.
* Navigate the site using the Default (Classic Blue) color scheme. The Sidebar and Masterbar should look identical color wise to what you see in production.

### CODE REVIEW

**Comment by frontdevde** [comment]
In the screenshots below, you can see the current state of the Sidebar and Masterbar in the PR for both Classic Bright and Laser Black. It's not done yet but given that the two color schemes aren't visible in production and how many files had to be touched already, I'd argue let's get this merged in the current stage and continue working on it on a new PR. If you agree @flootr @blowery, please remove the `In Progress` label and set it to `Needs Review`. Thank you!

screen shot 2018-12-05 at 23 59 50
screen shot 2018-12-05 at 23 59 21

File: assets/stylesheets/_main.scss [hunk 1]
```diff
@@ -70,6 +70,10 @@ body {
 	}
 }
 
+.layout {
+	background: var( --color-surface-backdrop );
```
Comment (frontdevde) [inline assets/stylesheets/_main.scss]: Added this variable to control the new background color behind surfaces that's introduced with Laser Black.

File: client/blocks/site/style.scss [hunk 2]
```diff
@@ -116,7 +116,7 @@
 	overflow: hidden;
 	white-space: nowrap;
 	&::after {
-		@include long-content-fade();
+		@include long-content-fade( $color: var( --sidebar-secondary-background-gradient ) );
```
Comment (flootr) [inline client/blocks/site/style.scss]: So this sass mixin does work with css custom properties? I wasn't sure about if it's possible.

File: client/components/keyed-suggestions/style.scss [hunk 3]
```diff
@@ -1,6 +1,10 @@
-/*
+/**
+ * /*
```
Comment (flootr) [inline client/components/keyed-suggestions/style.scss]: This comment doesn't look right, probably due to auto formatting

Comment (frontdevde) [reply client/components/keyed-suggestions/style.scss]: Yeah, looks like `prettier` being creative!

File: client/blocks/site/style.scss (hunk 2 above)
Comment (frontdevde) [reply client/blocks/site/style.scss]: Yes, I updated the mixin to work with CSS custom properties. You just need to make sure that the color value assigned to the property you pass in is wrapped in `hex-to-rgb()` function.

Comment (frontdevde) [reply client/blocks/site/style.scss]: Updated the documentation for the Mixin as well 👍

**Comment by frontdevde** [comment]
Good catch! Turns out that was due to an unescaped `lighten` function. Fixed!

**Comment by flootr** [comment]
Had another look and it seems there's still a minor difference between the current master and this branch. It's the background color of the buttons on the right to each navigation point:

screenshot 2018-12-10 at 11 19 43

(left is this branch, right is master)

I couldn't find anything else so we should be able to merge this PR after fixing this 👍

**Comment by frontdevde** [comment]
> Had another look and it seems there's still a minor difference between the current master and this branch. It's the background color of the buttons on the right to each navigation point:
> 
> I couldn't find anything else so we should be able to merge this PR after fixing this 👍

Pushed a change and also rebased with master 👍

**Comment by blowery** [comment]
> Code looks good to me although we should find a solution for /* @format */ pragmas in the long term.

We should drop them and always process with prettier. Or find a way to use `//` for the pragma instead of `/* */`

**Comment by vindl** [comment]
@frontdevde I think this PR has caused a regression in Gutenberg integration in Calypso:

screenshot 2018-12-12 at 13 18 59

Notice the gray background in the top part of the screen. You can also reproduce this on https://wpcalypso.wordpress.com/block-editor/post/.

**Comment by frontdevde** [comment]
@vindl Thanks, on it!

File: assets/stylesheets/_main.scss (hunk 1 above)
Comment (blowery) [reply assets/stylesheets/_main.scss]: this is the bit that broke gutenberg-in-calypso https://github.com/Automattic/wp-calypso/pull/29147#issuecomment-446569432

we do want to keep this, but we'll need to add a bgcolor override somewhere down the chain.

**Comment by Copons** [comment]
I think this also breaks the Classic Calypso editor @frontdevde @blowery 

screenshot 2018-12-12 at 16 58 10

**Comment by blowery** [comment]
Looking into it

**Pull request merged**